ElectronicObserver/Window/Tools/EquipmentUpgradePlanner/EquipmentUpgradePlanItemViewModel.cs

```diff
@@ -118,6 +122,7 @@ public EquipmentUpgradePlanItemViewModel(EquipmentUpgradePlanItemModel plan)
 		EquipmentPicker = Ioc.Default.GetService<EquipmentPickerService>()!;
 		TimeChangeService = Ioc.Default.GetService<TimeChangeService>()!;
```

Comment:
Ah, can you change those to GetRequiredService and remove the ! at the end?

Reply:
done